### PR TITLE
Fix linting, modify base OAI harvester

### PR DESCRIPTION
**Pull request by erinspace**
Linting now should reject missing URLs! Modified base classes to accept more customized base urls, and added optional time between requests for providers of different sensitivities. Also allowed for properties to be pulled from the OAI header as well as from within the dc elements.

### CODE REVIEW

File: scrapi/base/__init__.py [hunk 1]
```diff
@@ -89,16 +82,13 @@ def get_records(self, url, start_date, resump_token=''):
         print url
         data = requests.get(url)
 
-        try:
-            doc = etree.XML(data.content)
-        except etree.XMLSyntaxError:
-            return 'try a new url'
+        doc = etree.XML(data.content)
```
Comment (fabianvf) [inline scrapi/base/__init__.py]: Does this error not occur, or is it just handled elsewhere?

File: scrapi/base/__init__.py [hunk 2]
```diff
@@ -169,9 +159,8 @@ def get_properties(self, result, property_list):
 
         properties = {}
         for item in property_list:
```
Comment (fabianvf) [inline scrapi/base/__init__.py]: Would you mind commenting this block, just so it's more apparent why you are doing what it is you are doing?

File: scrapi/base/__init__.py (hunk 1 above)
Comment (erinspace) [inline scrapi/base/__init__.py]: Yep, error no longer occurs - handled it in a better way by just accepting longer URLs if necessary!

Comment (fabianvf) [inline scrapi/base/__init__.py]: ok, neat

File: scrapi/base/__init__.py [hunk 3]
```diff
@@ -169,9 +159,8 @@ def get_properties(self, result, property_list):
...
-            prop = (
-                result.xpath('//dc:{}/node()'.format(item), namespaces=self.NAMESPACES) or [''])
-
+            prop = (result.xpath('//dc:{}/node()'.format(item), namespaces=self.NAMESPACES) or [''])
+            prop += (result.xpath('//ns0:{}/node()'.format(item), namespaces=self.NAMESPACES) or [''])
             if len(prop) > 1:
```
Comment (fabianvf) [inline scrapi/base/__init__.py]: properties[item] = prop if len(prop) > 1 else prop[0]

**Comment by erinspace** [comment]
ready for second pass!

**Pull request merged**